src/modules/cmdpal/Microsoft.CmdPal.UI/App.xaml.cs

```diff
@@ -55,6 +59,13 @@ public App()
         Services = ConfigureServices();
 
         this.InitializeComponent();
+
+        NativeEventWaiter.WaitForEventLoop(
+            "Local\\PowerToysCmdPal-ExitEvent-eb73f6be-3f22-4b36-aee3-62924ba40bfd", () =>
```

Comment:
Do we _need_ this event to trigger the `Dispose` on the EtwTrace? If we just quit cmdpal, will that just dispose the `EtwTrace` automatically?

Reply:
The problem is with how PowerToys kills CmdPal on close. Calling TerminateProcess does not give a chance to cmdpal to dispose stuff